### PR TITLE
added some more details + debugging steps to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by janeyx99**
When attempting to install PyTorch locally on my Macbook, I had some difficulty running the setup steps and understanding what I was really doing. I've added some clarifications and summarized some debugging steps about `python setup.py develop` to lower the barrier of entrance for new contributors. 

I'm seeking a lot of review here since I am not sure if what I wrote is entirely the most useful or accurate. Thank you!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -118,11 +118,39 @@ For example:
...
+You do not need to repeatedly install after modifying Python files. Note that this is only true
+after modifying Python files and you would need to reinstall if you modify other files.
 
 In case you want to reinstall, make sure that you uninstall PyTorch first by running `pip uninstall torch`
 and `python setup.py clean`. Then you can install in `develop` mode again.
 
+### Tips and Debugging
+* A prerequisite to installing PyTorch is CMake. We recommend installing it with [Homebrew](https://brew.sh/)
+with `brew install cmake` if you are developing on MacOS or Linux system.
+* Our `setup.py` requires Python 3. Python 2 is deprecated anyway, so you should definitely get with the times
```
Comment (seemethere) [inline CONTRIBUTING.md:130]:
```suggestion
* Our `setup.py` requires Python >= 3.6
```

**Review by seemethere — COMMENTED** [review]
Looks mostly good to me, have some suggestions on some better wording

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -118,11 +118,39 @@ For example:
 - modify your Python file `torch/__init__.py`
 - test functionality
 
-You do not need to repeatedly install after modifying Python files.
+You do not need to repeatedly install after modifying Python files. Note that this is only true
+after modifying Python files and you would need to reinstall if you modify other files.
```
Comment (walterddr) [inline CONTRIBUTING.md:122]:
```suggestion
You do not need to repeatedly install after modifying Python files (`.py`). However, you would need to reinstall if you modify Python interface (`.pyi`, `.pyi.in`) or non-Python files (`.cpp`, `.cc`, `.cu`, `.h`, ...)
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -118,11 +118,39 @@ For example:
...
+  git submodule deinit -f .
+  git clean -xdf
+  python setup.py clean
+  git submodule update --init --recursive # very important to sync the submodules
+  python setup.py develop                 # then try running the command again
+  ```
+  3. If you still get failures, nuke your `build` directory. The `setup.py` script compiles binaries into the
+  `build` folder and caches many details along the way, which saves time the next time you build. If you're running
+  into issues, you can always `rm -rf build` from the toplevel `pytorch` directory and start over.
+  4. `python setup.py develop` is equivalent to running `make` from the `build` directory. If you want more detailed
```
Comment (walterddr) [inline CONTRIBUTING.md:147]: this is not true. I would suggest change it to:
```suggestion
  4. The main step within `python setup.py develop` is running `make` from the `build` directory. If you want more detailed
```

Although I don't think the command below gives more error message. you can also give environment variables via  `ENV_KEY1=ENV_VAL1[, ENV_KEY2=ENV_VAL2]* python setup.py develop`

**Review by walterddr — COMMENTED** [review]
looks great! thanks for updating the instructions. please take a look at the comments.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -118,11 +118,39 @@ For example:
...
+  2. If you have made edits to the PyTorch repo, commit any change you'd like to keep and clean the repo with the
+  following commands (note that clean _really_ removes all untracked files and changes.):
+  ```bash
+  git submodule deinit -f .
+  git clean -xdf
+  python setup.py clean
+  git submodule update --init --recursive # very important to sync the submodules
+  python setup.py develop                 # then try running the command again
+  ```
+  3. If you still get failures, nuke your `build` directory. The `setup.py` script compiles binaries into the
```
Comment (samestep) [inline CONTRIBUTING.md:144]: I'm probably missing something, but doesn't `git clean -xdf` already remove the `build` directory, making this step redundant?
```
$ git clean -xdn | grep build
Would remove build/
```

**Review by samestep — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 4 above)
Comment (janeyx99) [reply CONTRIBUTING.md:143]: True, I'll swap steps 2 and 3

**Review by samestep — APPROVED** [review]

**Pull request closed without merging**